subprojects/gradle/build-trace/src/main/kotlin/com/avito/android/build_trace/internal/critical_path/CriticalPathListener.kt

```diff
@@ -0,0 +1,47 @@
+package com.avito.android.build_trace.internal.critical_path
+
+import com.avito.android.gradle.metric.AbstractBuildEventsListener
+import com.avito.android.gradle.profile.BuildProfile
+import com.avito.android.gradle.profile.TaskExecution
+import com.avito.graph.ShortestPath
+import org.gradle.BuildResult
+import org.gradle.api.Task
+
+internal class CriticalPathListener(
+    private val serialization: CriticalPathSerialization
+) : AbstractBuildEventsListener(), CriticalPathProvider {
+
```

Comment:
This is a provider for the critical path. It converts executed tasks to the algorithm.